### ISSUE TITLE
Change `require` to always load as CommonJS and `import` always as ESM

### ISSUE THREAD

**Issue by fatcerberus** [spheredev/neosphere#287]
We'll call this "Phase 1" of #280. A more comprehensive change may be in order for miniSphere 6.0, so this might not be the end of the story.  Time will tell.

The semantics of `require()` will be changed as follows:
* Filenames ending in `.mjs` will be removed from the module search patterns
* The file extension will be ignored and modules will always be loaded as CommonJS
* Loading ESM modules synchronously using `require()` will no longer be possible.  ESM is supposed to be fully async, so while allowing this in a host-defined function probably doesn't violate the letter of the specification, it does violate the spirit of the text.

The semantics of `import`, both dynamic and static, will be changed as follows:
* Node.js-like patterns (at least `specifier/package.json`) will be removed from the module search.
* The file extension will be ignored and modules will always be loaded as ESM.
* It will no longer be possible to `import` CommonJS modules.  `require` should be used instead as that's the only reason it even exists.

It might be possible in the future to restore the CommonJS-ESM cross-compatibility in some capacity, but it's best to draw a strict line now to avoid creating unnecessary BC baggage.  miniSphere is not yet too widely used, so 5.3 may be our last chance to make this change.

**Comment by fatcerberus** [comment]
I'm hoping I don't have to use that `breaking change` label very often.  The "API freeze" concept exists for a reason.

That said, `require()` is not covered by API versioning and `import` silently loading anything as CommonJS could reasonably be considered a bug, so I'm not too concerned in this particular case.  I'm still going to use the tag out of principle, though.

**Comment by fatcerberus** [comment]
Adding `refactor` tag since all module loading currently goes through the same channels, so some code will need to be changed in order to differentiate `require` from `import`.

**Comment by fatcerberus** [comment]
I made a small compromise: `require()` can still load a script as ESM code *if* `FS.typeOf(fileName)` is "script/es-module" (or in Cell, if the filename ends in `.mjs`).  `script/javascript` therefore becomes the "adaptive" file type for which loading semantics depend on the function doing the loading.  The reason I did it this way is that, thanks to Node, `.mjs` is unambiguously an ES module so it didn't make much sense to try to load such scripts as CommonJS.

The important thing is that `import` never attempt to load a module as CommonJS and avoid finding known-to-be-CommonJS things like `package.json` when searching for a module.  Both of those changes were implemented in a1b9b5b386920fa341c9f175d1117a83962a6298 so this issue can be closed.
